### PR TITLE
New action for due date

### PR DESCRIPTION
**Pull request by colttt**
Do you follow the guidelines?

- [x] I have tested my changes
- [x] There is no breaking change
- [x] There is no regression
- [x] I follow existing [coding style](https://docs.kanboard.org/en/latest/developer_guide/coding_standards.html)

it should fix https://github.com/kanboard/kanboard/issues/4741

### CODE REVIEW

File: app/Action/TaskAssignDueDateOnMoveColumn.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+     * Get the required parameter for the action (defined by the user)
+     *
+     * @access public
+     * @return array
+     */
+    public function getActionRequiredParameters()
+    {
+        return array(
+            'duration' => t('Duration in days'),
+                                                'column_id' => t('Column'),
```
Comment (fguillot) [inline app/Action/TaskAssignDueDateOnMoveColumn.php:49]: The indentation seems a little bit off.

File: app/Action/TaskAssignDueDateOnMoveColumn.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        );
+    }
+/**
```
Comment (fguillot) [inline app/Action/TaskAssignDueDateOnMoveColumn.php:52]: Same here

File: app/Action/TaskAssignDueDateOnMoveColumn.php [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+     * @return string[]
+     */
+    public function getEventRequiredParameters()
+    {
+        return array(
+            'task_id',
+            'task' => array(
+                'project_id',
+            ),
+                                                'src_column_id',
```
Comment (fguillot) [inline app/Action/TaskAssignDueDateOnMoveColumn.php:65]: Same here

**Review by fguillot — COMMENTED** [review]
It would be nice to add a unit test for your automated action.

**Comment by colttt** [comment]
I fixed the indentation.

>It would be nice to add a unit test for your automated action.

I just used other actions as a template, so what do you mean exactly?

**Comment by fguillot** [comment]
Look here: https://github.com/kanboard/kanboard/tree/master/tests/units/Action

**Pull request merged**